src/avt/Queries/Queries/avtXRayImageQuery.C

```diff
@@ -315,6 +328,25 @@ avtXRayImageQuery::SetInputParams(const MapNode &params)
         SetBackgroundIntensities(v);
     }
 
+    if (params.HasNumericVectorEntry("energy_group_bins"))
+    {
+        doubleVector v;
+        params.GetEntry("energy_group_bins")->ToDoubleVector(v);
+        SetEnergyGroupBins(v);
+    }
+
+    // Are you ever going to have just one energy group bin? No.
+    // But this is here for helpful error messaging. It is possible
+    // to pass just one number in under energy_group_bins, so with
+    // this logic here VisIt will give users sensible error messages
+    // embedded within the blueprint metadata.
+    if (params.HasNumericEntry("energy_group_bins"))
```

Comment:
I'm not sure I understand this comment. It mentions helpful error messaging, but I don't see any tests on the count or error message printing.

Reply:
I embedded a warning/error message in the blueprint metadata.

So, in the location you'd normally see your energy group bins come out, you get a string instead that explains why you didn't get the energy group bins.

Reply:
Ah, that makes sense. Sounds good.